File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_len , left = 0, 0
        seen_till_now = set()

        for right, char in enumerate(s):
            while char in seen_till_now:
                seen_till_now.remove(s[left])
                left += 1
            seen_till_now.add(char)
            max_len = max(max_len, right - left + 1)
        
        return max_len
        

File: Data Structures & Algorithms/max-water-container/submission-3.java
class Solution {
    public int maxArea(int[] heights) {
        int left = 0, right = heights.length - 1;

        int maxArea = 0;

        while(right > left){
            int currArea = (right - left) * Math.min(heights[left], heights[right]);
            maxArea = Math.max(maxArea, currArea);

            if(heights[left] > heights[right]){
                right--;
            }else{
                left++;
            }
        }

        return maxArea;
    }
}
